app/javascript/components/widgets/numbers/falling-blocks.jsx

```diff
@@ -159,24 +195,61 @@ class Scene extends React.Component {
     // randomly pick a point from one wall edge to the other
     // so that they don't fall directly on top of each other.
     if (block) {
-      this.setState(state => ({ [block]: state[block] + 1 }));
-      World.add(
-        this.engine.world,
-        // eslint-disable-next-line prettier/prettier
-        Bodies.polygon(randomPositionBetweenWalls, -10, 8, 8, {
-          restitution: 0.25,
-          friction: 0.8,
-          render: {
-            sprite: {
-              texture: blockTypes[block],
-            },
+      const newBlock = Bodies.polygon(randomPositionBetweenWalls, -10, 8, 8, {
+        restitution: 0.25,
+        friction: 0.8,
+        render: {
+          sprite: {
+            texture: blockTypes[block],
           },
-        }),
-      );
+        },
+      });
+
+      this.setState(state => ({ [block]: state[block] + 1 }));
+
+      World.add(this.engine.world, newBlock);
+
+      Events.on(newBlock, 'sleepStart', body => this.addToOverlay(body.source));
     }
     this.timer = setTimeout(this.addBlocks, 200);
   };
 
+  addToOverlay(body) {
+    if (!body || !body.render.sprite.texture || body.isOnOverlay) {
+      return;
+    }
+    invisify(body);
+    setStatic(body);
+    const ctx = this.overlay.current.getContext('2d');
+    const { texture } = body.render.sprite;
+    const { sprite } = body.render;
+    ctx.translate(body.position.x, body.position.y);
+    ctx.rotate(body.angle);
+
+    const img = new Image();
+    img.src = texture;
+    ctx.drawImage(
+      img,
+      img.width * -sprite.xOffset,
+      img.height * -sprite.yOffset,
+    );
+
+    ctx.rotate(-body.angle);
+    ctx.translate(-body.position.x, -body.position.y);
+    const b = body;
+    b.isOnOverlay = true;
+    Events.off(body, 'sleepStart', overlayBody =>
```

Comment:
not sure Events.off gets a callback

Reply:
Looks like it:
    /**
     * Removes the given event callback. If no callback, clears all callbacks in `eventNames`. If no `eventNames`, clears all events.
     * @method off
     * @param {} object
     * @param {string} eventNames
     * @param {function} callback
     */
`    Events.off = function(object, eventNames, callback) {`

https://github.com/liabru/matter-js/blob/2ec247b7af1c6b5da6ee05c73274ed5822c73503/src/core/Events.js#L45

Reply:
do we need to `addToOverlay` on this callback?

Reply:
nope, it's gone now